### PR TITLE
rgw: fix cls_bucket_head result order consistency

### PR DESCRIPTION
**Pull request by tianshan**
headers's key is string but bucket_instance_ids's key is int
looked around and found no one use header's key, so change that

Signed-off-by: Tianshan Qu <tianshan@xsky.com>

### CODE REVIEW

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -13017,7 +13017,7 @@ int RGWRados::cls_bucket_head(const RGWBucketInfo& bucket_info, int shard_id, ma
 
   map<int, struct rgw_cls_list_ret>::iterator iter = list_results.begin();
   for(; iter != list_results.end(); ++iter) {
-    headers[oids[iter->first]] = iter->second.dir.header;
+    headers[iter->first] = iter->second.dir.header;
```
Comment (cbodley) [inline src/rgw/rgw_rados.cc]: since `list_results` is a local variable, we can use `std::move(iter->second.dir.header)` here instead

**Comment by tianshan** [comment]
looks like vector is enough, I will fix that

**Comment by tianshan** [comment]
@cbodley  done, please review

**Comment by yuriw** [comment]
[wip-yuri-testing-2017-11-15-1751](https://trello.com/c/TVbI0GDr/202-wip-yuri-testing-2017-11-15-1751)

**Comment by cbodley** [comment]
i've opened a luminous backport at https://github.com/ceph/ceph/pull/26546

**Pull request merged**